### PR TITLE
Run initContainer in privileged mode for cilium cleanup

### PR DESCRIPTION
**Pull request by abhide15**
Tickets: GH-4961

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4994)

### CODE REVIEW

**Review by tgraf — APPROVED** [review]

**Review by rlenglet — APPROVED** [review]
If you're using `privileged: true`, I think you shouldn't need to add explicitly any capabilities.

**Comment by tgraf** [comment]
test-me-please

**Comment by abhide15** [comment]
@joestringer do we need NET_ADMIN capability?
For me, the cleanup works even without privileged: True or omitting NET_ADMIN capability or omitting both.

**Comment by aanm** [comment]
Please don't merge this PR until this one is merged first https://github.com/cilium/cilium/pull/4849

**Comment by joestringer** [comment]
@abhide both are likely required in some environments, privileged for the filesystem path access and CAP_NET_ADMIN for bpf map delete.. just not sure whether privileged implies CAP_NET_ADMIN.

**Comment by ianvernon** [comment]
@abhide15 please rebase, and also sign-off on your commit per [Cilium's Contributing Documentation for submitting a pull request](http://docs.cilium.io/en/latest/contributing/#submitting-a-pull-request).

Also, in the above link, please ensure that your PR meets the requirements established by steps 4, 5, and 6. With respect to step 4, please update the commit description illustrating the problem that this fixes; this makes it easier to understand why things were changed when viewing the Git history. Feel free to message me offline if you have any questions about these steps.

File: examples/kubernetes/1.7/cilium-crio.yaml
Comment (abhide15) [inline examples/kubernetes/1.7/cilium-crio.yaml]: @aanm are these changes expected?

Comment (aanm) [reply examples/kubernetes/1.7/cilium-crio.yaml]: @abhide15 please ignore those changes in 1.7 manually for now. We will drop support for k8s 1.7 in 1.3

Comment (abhide15) [reply examples/kubernetes/1.7/cilium-crio.yaml]: Fixed

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
@tgraf  - @abhide15 addressed @aanm 's comments, and since @aanm is out I think we can merge this.

**Comment by tgraf** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**
